RestaurantReservation.Db/Repositories/CustomerRepository.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Microsoft.EntityFrameworkCore;
+using RestaurantReservation.Db.DataModels;
+using RestaurantReservation.Db.Utilities;
+
+namespace RestaurantReservation.Db.Repositories
+{
+    public class CustomerRepository : IRepositoryServices<Customer,string>
+    {
+        private readonly RestaurantReservationDbContext _dbContext;
+
+        public CustomerRepository(RestaurantReservationDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public async Task<OperationResult<string>> AddAsync(Customer customer)
+        {
+            try
+            {
+                await _dbContext.Customers.AddAsync(customer);
+                await _dbContext.SaveChangesAsync();
+
+                return OperationResult<string>.SuccessResult($"id:{customer.Id}");
+            }
+            catch (Exception ex)
+            {
+                return OperationResult<string>.FailureResult($"Failed to add customer. {ex.Message}");
+            }
+        }
+
+        public async Task<bool> DeleteAsync(int id)
+        {
+            var customerToDelete = await _dbContext.Customers.FindAsync(id);
+            if(customerToDelete != null)
+            {
+                 _dbContext.Customers.Remove(customerToDelete);
+                await _dbContext.SaveChangesAsync();
+                return true;
+            }
+            return false;
+        }
+
+        public async Task<IEnumerable<Customer>> GetAllAsync()
+        {
+            return await _dbContext.Customers.ToListAsync();    
+        }
+
+        public async Task<Customer?> GetByIdAsync(int id)
+        {
+            return await _dbContext.Customers.FindAsync(id);
+        }
+
+        public async Task<OperationResult<string>> UpdateAsync(int id, Customer updatedCustomer)
+        {
+            var rowsEffected = await _dbContext.Customers.Where(customer => customer.Id == id)
```

Comment:
you could've returned the entity and update the fields and then saved the changes, let's discuss this.